src/persist/src/operators/stream.rs

```diff
@@ -303,6 +308,57 @@ where
     }
 }
 
+/// Extension trait for [`Stream`].
+pub trait AwaitFrontier<G: Scope<Timestamp = u64>, D> {
+    /// Stashes data until it is no longer beyond the input frontier.
+    ///
+    /// This is similar, in spirit, to what `consolidate()` does for differential collections and
+    /// what `delay()` does for timely streams. However, we are not working with differential
+    /// collections and `delay()` deals with changing the timestamp while the behaviour we want is to
+    /// wait for the frontier to pass. The latter is an implementation detail of `delay()` that is not
+    /// advertised in its documentation.
+    fn await_frontier(&self, name: &str) -> Stream<G, (D, u64, isize)>;
```

Comment:
nit: is it maybe slightly more idiomatic to express this in terms of `(D, Scope::Timestamp, isize)` - genuine question not sure of myself.

Reply:
It probably would be, but all the existing operators in `persist` hard-code `u64`. So I think if we want to change it we should do one big refactoring instead of doing a different thing for this operator. WDYT?